### PR TITLE
OPSEXP-2121 Add kics scan

### PR DESCRIPTION
**Pull request by pmacius**
Ref: OPSEXP-2121

### CODE REVIEW

File: .github/workflows/kics.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  kics:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: run kics Scan
```
Comment (gionn) [inline .github/workflows/kics.yml:28]:
```suggestion
      - name: Run KICS Scan
```

File: .github/workflows/kics.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        uses: checkmarx/kics-github-action@530ac1f8efe6202b0f12c9a6e952597ae707b755 # v2.1.2
+        with:
+          path: 'playbooks,roles'
+          ignore_on_exit: results
+          output_path: report-dir/
+          output_formats: 'sarif'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          enable_jobs_summary: true
```
Comment (gionn) [inline .github/workflows/kics.yml:36]:
```suggestion
          enable_jobs_summary: true
          enable_comments: false
```

we agreed on that right?

File: .github/workflows/kics.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  kics:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: run kics Scan
+        uses: checkmarx/kics-github-action@530ac1f8efe6202b0f12c9a6e952597ae707b755 # v2.1.2
+        with:
+          path: 'playbooks,roles'
```
Comment (gionn) [inline .github/workflows/kics.yml:31]: maybe also add `platform_type: ansible` if make the scan faster?

**Review by gionn — CHANGES_REQUESTED** [review]
we could also disable secret scanning with `disable_secrets` given that we are already using detect-secrets in pre-commit

File: .github/workflows/kics.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: kics
+
+on:
+  pull_request:
+    branches: [master]
+    paths-ignore:
+      - "docs/**"
+      - "*.md"
+      - .github/workflows/docs.yml
+      - .pre-commit-config.yaml
```
Comment (gionn) [inline .github/workflows/kics.yml:10]: given you have `path: 'playbooks,roles'` I would add the same as paths here

File: .github/workflows/kics.yml (hunk 2 above)
Comment (pmacius) [reply .github/workflows/kics.yml:36]: enable_comments defaults to false

Comment (gionn) [reply .github/workflows/kics.yml:35]: https://github.com/Checkmarx/kics-github-action?tab=readme-ov-file#inputs

I was assuming the first column was `default` 🤦🏻

File: .github/workflows/kics.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      - 'playbooks/**'
+      - 'roles/**'
+      - '.github/workflows/kics.yml'
+  push:
+    branches: [master]
+    paths-ignore:
+      - "docs/**"
+      - "*.md"
+      - .github/workflows/docs.yml
+      - .pre-commit-config.yaml
```
Comment (gionn) [inline .github/workflows/kics.yml:16]: same as before for consistency

File: .github/workflows/kics.yml (hunk 2 above)
Comment (pmacius) [reply .github/workflows/kics.yml:35]: Yeah not very intuitive - also the column with default values is hidden a little :D

File: .github/workflows/kics.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+permissions:
+  pull-requests: write
+  contents: write
```
Comment (gionn) [inline .github/workflows/kics.yml:20]: are those still required with comments disabled?

Comment (pmacius) [reply .github/workflows/kics.yml:20]: Ahh right let me test not sure - probably contents still needs to be there to upload sarif.

Comment (pmacius) [reply .github/workflows/kics.yml]: not permission required

**Review by gionn — APPROVED** [review]

**Pull request merged**